test/streams.js

```diff
@@ -45,3 +48,87 @@ describe('strong-remoting', function() {
       .expect({bar: 'baz'}, done);
   });
 });
+
+describe('a function returning a ReadableStream', function() {
+  var Readable = require('stream').Readable;
+  var remotes = RemoteObjects.create();
+  var streamClass;
+  var server;
+
+  before(function(done) {
+    var test = this;
+    // NOTE: Date object will not be coerced back into a Date
+    var data = test.data = [{foo: 'bar'}, 'bat', false, 0, null];
+    var app = express();
+    server = app.listen(0, '127.0.0.1', done);
+    function StreamClass() {
+
+    }
+
+    StreamClass.createStream = function createStream(cb) {
+      cb(null, es.readArray(test.data));
+    };
+
+    StreamClass.createStreamWithError = function createStreamWithError(cb) {
+      var rs = new Readable({objectMode: true});
+      rs._read = function() {
+        rs.emit('error', new Error('test error'));
+      };
+      cb(null, rs);
+    };
+
+    streamClass = new SharedClass('StreamClass', StreamClass);
+
+    streamClass.defineMethod('createStream', {
+      isStatic: true,
+      fn: StreamClass.createStream,
+      returns: [{
+        arg: 'stream',
+        type: 'ReadableStream',
+        json: true
```

Comment:
Should we specify `root:true` as well? Or is that implied by `type:ReadableStream`? Could you please use a different arg name in the test, just to make sure that we don't accidentally match it when deciding whether the argument is of a stream type?